ui/src/app/store/subnet/types/actions.ts

```diff
@@ -21,6 +21,20 @@ export type CreateParams = {
   vlan?: Subnet["vlan"];
 };
 
-export type UpdateParams = CreateParams & {
+export type UpdateParams = {
   [SubnetMeta.PK]: Subnet[SubnetMeta.PK];
+  active_discovery?: Subnet["active_discovery"];
+  allow_dns?: Subnet["allow_dns"];
+  allow_proxy?: Subnet["allow_proxy"];
+  cidr?: Subnet["cidr"];
+  description?: Subnet["description"];
+  disabled_boot_architectures?: string;
+  dns_servers?: Subnet["dns_servers"];
+  fabric?: Fabric[FabricMeta.PK];
+  gateway_ip?: Subnet["gateway_ip"];
+  managed?: Subnet["managed"];
+  name?: Subnet["name"];
+  rdns_mode?: Subnet["rdns_mode"];
+  vid?: VLAN["vid"];
+  vlan?: Subnet["vlan"];
```

Comment:
@huwshimi I think we might need to go through these UpdateParams because most of the time we only need the id then the rest are optional.

Reply:
Sure, maybe this is a case for union types that match what the API expects? Although, looking at the subnet form it looks like there are [a lot of different related fields](https://github.com/maas/maas/blob/6d3edc8baffc56fad32c58ad6fecda3b0e19a966/src/maasserver/forms/subnet.py#L143) so maybe we'd introduce bugs when trying to mirror that here...